main/zoe/api/Zoe-Intro.md

```diff
@@ -0,0 +1,109 @@
+
+
+Zoe has two distinct APIs. One is client/user-facing, with `zoe.` prefixes on its commands. The other is only exposed to contracts built on Zoe (its *contract facet*), with `zcf.` prefixes on its commands. . 
+
+
+## Starting a contract on Zoe
+
+
+While there can be more than one Zoe instance installed and running at any time, for now we’ll focus on the single instance running on Agoric’s testnet. Of course, before you can run a smart contract on Zoe, you have to install it there. For Zoe purposes, *installation* means installing the code your contract is written in (JavaScript and of course the ERTP and Zoe API commands we’ve defined) on a Zoe instance. 
+
+
+Contract installation returns an `installationHandle`, a unique identifier for the installed contract used to create *contract instances* from the installed code. A contract installation is a way to reuse the same code multiple times and ensure that widely used contract code is recognizable under the same identity. For example, if I negotiate with someone and we agree to use the “Atomicswap” contract **tyg todo link**, if I make an instance of that contract, my counterparty can use its `installationHandle` with the `getInstallation()` API method to check the code is the `Atomicswap` code they expected.
+
+
+## Running contracts on Zoe
+
+
+First, use `zoe.makeInstance()` to create an instance of the installed contract you want to use. The method takes as arguments:
+- the `installationHandle` of the specific contract to use.
+- a *keyword record* of name : value pairs.
+  - The names are words used in the contract, such as “Asset” or “Bid” that, in the contract, take an amount as its value. 
+  - The values are references to an *issuer* (see Introduction to ERTP **tyg todo: add link) of the type you want the associated name to accept as its value in the contract instance. For example, if your contract uses the name “Asset” for what the offer in the contract is (e.g. 3) and the resource unit is quatloos, and uses “Price” for the counter-offer, which is in moola, the keyword record value would be ((“Asset” : quatlooIssuer) (“Price” : moolaIssuer)).
+- *terms*: Another set of key : value pairs.
+  -   These customize contract aspects expressed in the contract without specific values. For example, an auction contract’s may not define what the minimum bid has to be or the minimum amount one has to raise from the current high bid, just that there are such values which the contract uses. These values are different for a collection of used Harry Potter paperbacks and a signed first British hardback edition of *Harry Potter and the Philosopher’s Stone*. If you write code in the contract to access that instance’s `terms`’ name:value pairs of `('MinBid' : 25, 'Currency : 'USDollar', 'MinRaise' : 5)` or  `('MinBid' : 100000, 'Currency : 'USDollar', 'MinRaise' : 5000)`, you’ve customized the generic contract for that specific instance. 
```

Comment:
parens should be curlies again. Don't need to quote keywords since they are (and must be) identifiers.

Reply:
Was doing an abstract example, but you're right that showing what would actually be there is better. Will correct.

Reply:
> These customize contract aspects expressed in the contract without specific values

I think this is too fuzzy. maybe: "Terms allow the creator of a contract Instance to further customize the contract operations, as enabled by the contract code."

> For example, an auction contract’s may not define what the minimum bid has to be or the minimum amount one has to raise from the current high bid, just that there are such values which the contract uses.

Maybe instead: "For example, the code of an auction contract might allow the instantiator to decide the minimum bid, or the minimum amount one has to raise from the current high bid, or the number of bids before the auction closes."

I think the used Harry Potter paperback and signed first edition is a great example of why you might want a different minimum bid. The start of the sentence is confusing to me - maybe instead of "these values are different", it should be "Why do we need to be able to parameterize a contract with terms? Well, our minimum bid would be quite different if the auction is for a used Harry Potter paperback vs a signed first edition..."

Reply:
By "minimum bid" do we mean the reserve price?

Normally, an auction would use the seller's `want:` as the reserve price.